parser_library/test/workspace/b4g_integration_test.cpp

```diff
@@ -200,67 +226,108 @@ class pgm_conf_preference_helper
+    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p1_mac1.get_uri(), sys_sub_p2_mac1.get_uri() }));
 
-    run_if_valid(ws.did_open_file(pgm_a));
-    parse_all_files(ws);
-    ws.collect_diags();
+    change_reparse_and_recollect_diags(
+        fm, ws, pgm_conf_rl, std::regex_replace(helper.pgm_conf_template, std::regex("\\$x"), "*"));
+    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p1_mac1.get_uri(), sys_sub_p1_mac1.get_uri() }));
 
-    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p1_mac1.get_uri(), sys_sub_p2_mac1.get_uri() }));
+    change_reparse_and_recollect_diags(
+        fm, ws, pgm_conf_rl, std::regex_replace(helper.pgm_conf_template, std::regex("\\$x"), "DIFFERENT_FILE"));
+    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p2_mac1.get_uri(), sys_sub_p2_mac1.get_uri() }));
+
+    change_reparse_and_recollect_diags(fm,
+        ws,
+        b4g_conf_rl,
+        R"({"elements":{"B":{"processorGroup":"P2"}},"defaultProcessorGroup":"P3","fileExtension":""})");
+    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p2_mac1.get_uri(), sys_sub_p3_mac1.get_uri() }));
 }
 
-TEST(b4g_integration_test, pgm_conf_preference_regex_path)
+TEST(b4g_integration_test, configuration_preference_missing_proc_groups_alternatives)
 {
-    pgm_conf_preference_helper helper("A");
+    missing_proc_grps_helper helper("A");
+
     auto& ws = helper.ws;
     auto& fm = helper.fm;
 
-    run_if_valid(ws.did_open_file(pgm_b));
-    parse_all_files(ws);
-    ws.collect_diags();
+    open_parse_and_recollect_diags(ws, { pgm_a });
+    EXPECT_TRUE(matches_message_text(ws.diags(), { diagnostic_s::error_W0004(pgm_a, "NON_EXISTENT_PGM").message }));
 
-    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p2_mac1.get_uri() }));
+    change_reparse_and_recollect_diags(
+        fm, ws, pgm_conf_rl, std::regex_replace(helper.pgm_conf_template, std::regex("\\$x"), "*"));
+    EXPECT_TRUE(matches_message_text(ws.diags(), { diagnostic_s::error_W0004(pgm_a, "NON_EXISTENT_PGM").message }));
 
-    ws.diags().clear();
-    change_and_reparse(fm, ws, pgm_conf_rl, std::regex_replace(helper.pgm_conf_template, std::regex("\\$x"), "*"));
+    change_reparse_and_recollect_diags(
+        fm, ws, pgm_conf_rl, std::regex_replace(helper.pgm_conf_template, std::regex("\\$x"), "DIFFERENT_FILE"));
+    EXPECT_TRUE(matches_message_text(ws.diags(), { diagnostic_s::error_B4G002(pgm_a, "NON_EXISTENT_B4G").message }));
 
-    run_if_valid(ws.did_open_file(pgm_a));
-    parse_all_files(ws);
-    ws.collect_diags();
-
-    EXPECT_TRUE(matches_message_text(ws.diags(), { sys_sub_p1_mac1.get_uri(), sys_sub_p1_mac1.get_uri() }));
+    change_reparse_and_recollect_diags(
+        fm, ws, b4g_conf_rl, std::regex_replace(helper.b4g_conf_template, std::regex("\\$x"), "DIFFERENT_FILE"));
+    EXPECT_TRUE(
+        matches_message_text(ws.diags(), { diagnostic_s::error_B4G002(pgm_a, "NON_EXISTENT_B4G_DEFAULT").message }));
```

Comment:
Done. 

I also tried to create something like `contains_partial_msg_text`. However, it would require to use something different than `std::includes` as it expects that the provided predicate is used also during sorting.